include/screen_capture/ScreenCapture.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <windows.h>
+#include <iostream>
+
+namespace OutputCapture
+{
+    class ScreenCapture
+    {
+        private:
+            ScreenCapture(const ScreenCapture &) = delete;
+            ScreenCapture(const ScreenCapture &&) = delete;
+            const ScreenCapture &operator=(const ScreenCapture &) = delete;
+            const ScreenCapture &&operator=(const ScreenCapture &&) = delete;
+
+            static ScreenCapture *pInstance;
+
+            ScreenCapture();
+
+            bool SaveBMPFile(const char *filename, HDC bitmapDC, uint16_t width, uint16_t height) const;
+            bool StartCapture(uint16_t x, uint16_t y, uint16_t width, uint16_t height, const char *filename) const;
+            const std::tuple<uint16_t, uint16_t> GetDesktopResolution() const;
+            int GetFilePointer(HANDLE FileHandle) const;
+
+        public:
+            ~ScreenCapture();
+
+
+            static ScreenCapture *GetOrCreateInstance()
+            {
+                if (!pInstance)
+                {
+                    pInstance = new ScreenCapture;
+                }
+                return pInstance;
+            }
+
+            static void DeleteInstance()
+            {
+                if (pInstance)
+                {
+                    delete pInstance;
+                }
+            }
+
+            bool CaptureFullScreen(std::string outputFileName) const;

```

Comment:
if this function return a bool, you should named it by IsCaptureFullScreenSucessfull or something like this . because when i'm reading this function i can't guess that this function return a bool (except if i read the code :p )

Reply:
In fact this function is not here to provide a state but to do a real action. But I do agree that returning a boolean here is pretty misleading. What do you think about throwing an error when an issue occurs ?

Reply:
In addition tout that you forgot the const ref on your std::string